web/api/v1/api.go

```diff
@@ -127,6 +130,59 @@ func (api *API) options(r *http.Request) (interface{}, *apiError) {
 	return nil, nil
 }
 
+// TODO(jsanford): FIXME: This is a sample implementation.
+func (api *API) queryRule(r *http.Request) (interface{}, *apiError) {
+	var ts model.Time
+	if t := r.FormValue("time"); t != "" {
+		var err error
+		ts, err = parseTime(t)
+		if err != nil {
+			return nil, &apiError{errorBadData, err}
+		}
+	} else {
+		ts = api.now()
+	}
+
+	st, err := promql.ParseStmts(string(r.FormValue("query")))
+	if err != nil {
+		return nil, &apiError{errorBadData, err}
+	}
+
+	rls, err := rules.NewRules(st)
+	if err != nil {
+		return nil, &apiError{errorBadData, err}
+	}
+
+	var v []interface{}
+	for _, r := range rls {
+		vector, err := r.Eval(ts, api.QueryEngine)
+		if err != nil {
+			switch err.(type) {
+			case promql.ErrQueryCanceled:
+				return nil, &apiError{errorCanceled, err}
+			case promql.ErrQueryTimeout:
+				return nil, &apiError{errorTimeout, err}
+			}
+			return nil, &apiError{errorExec, err}
+		}
+
+		// TODO(jsanford): Do we want to return this at all?  Probably not.
```

Comment:
I was thinking that it'd be useful to be able to provide guidance on what FOR to use, rather than testing just one value. The standard start/end/step would be over say a week and then you'd get output (which could be via JS) saying there were N series lasting M minutes etc.